g_docs.go

```diff
@@ -1245,6 +1249,39 @@ func generatePathInfo(file *ast.File) (map[string]string, error) {
 	return pathInfo, nil
 }
 
+func getGoFilesInPackage(pkg string) (map[string]*ast.Package, error) {
+	if isPackageIgnored(pkg) {
+		return nil, nil
+	}
+
+	fileSet := token.NewFileSet()
+	astPkgs, err := parser.ParseDir(fileSet, pkg, func(info os.FileInfo) bool {
+		name := info.Name()
+		return !info.IsDir() &&
+			strings.HasSuffix(name, ".go") &&
+			!strings.HasPrefix(name, ".")
+	}, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+
+	return astPkgs, nil
+}
+
+func isPackageIgnored(pkg string) bool {
+	curPath, err := defaultOS.Getwd()
+	if err != nil {
+		return false
+	}
+	handlersPath := path.Join(curPath, "/handlers")
```

Comment:
The ticket desc indicates we will use it on microservices. Is this true? will we use swag or `bee` on other microservices? Because defined folder `/handlers` like this won't work on microservice `/pkg/router`.

> Currently, Zalora's microservices lack comprehensive and automated API documentation generation, which hinders efficient collaboration and integration efforts among development teams.

Reply:
No. 
> To address this challenge, we propose adopting the Swaggo Swagger Generator ([GitHub - swaggo/swag: Automatically generate RESTful API documentation with Swagger 2.0 for Go.](https://github.com/swaggo/swag) ) to automate the generation of Swagger documentation for all microservices within the Zalora ecosystem including but not limited to Doraemon and Costa services.

This PR exists so that `beego` doesn't check for declarative comments in chi endpoints' handlers. In other words, bee wouldn't be used in all things related to chi endpoints including the microservices 🙂

Reply:
@adirizka7 ....  "no" refers to which part?

> The ticket desc indicates we will use it on microservices. Is this true?
> to automate the generation of Swagger documentation for all microservices within the Zalora ecosystem including but not limited to Doraemon and Costa services

or 

> Because defined folder `/handlers` like this won't work on microservice `/pkg/router.`

Regardless, I can see what you achieve on this ticket.
I'll mark it as LGTM

Reply:
Ah sorry to confuse you, let me clarify.

> The ticket desc indicates we will use it on microservices.
Is this true?

No, not true, because in the ticket desc, it is mentioned that we'll use `swag` not `bee`.

> will we use swag or bee on other microservices? Because defined folder /handlers like this won't work on microservice /pkg/router.

As mentioned above, we'll not use bee, so it doesn't matter if it works or not. Hope this clarifies 🙇